src/Compilers/CSharp/Portable/BoundTree/BoundBinaryOperator.UncommonData.cs

```diff
@@ -24,18 +42,32 @@ internal class UncommonData
             public readonly ConstantValue? ConstantValue;
             public readonly MethodSymbol? Method;
             public readonly TypeSymbol? ConstrainedToType;
+            public readonly bool IsUnconvertedInterpolatedStringAddition;
+            public readonly InterpolatedStringHandlerData? InterpolatedStringHandlerData;
 
             // The set of method symbols from which this operator's method was chosen.
             // Only kept in the tree if the operator was an error and overload resolution
             // was unable to choose a best method.
             public readonly ImmutableArray<MethodSymbol> OriginalUserDefinedOperatorsOpt;
 
-            private UncommonData(ConstantValue? constantValue, MethodSymbol? method, TypeSymbol? constrainedToType, ImmutableArray<MethodSymbol> originalUserDefinedOperatorsOpt)
+            private UncommonData(ConstantValue? constantValue, MethodSymbol? method, TypeSymbol? constrainedToType, ImmutableArray<MethodSymbol> originalUserDefinedOperatorsOpt, bool isUnconvertedInterpolatedStringAddition, InterpolatedStringHandlerData? interpolatedStringHandlerData)
             {
                 ConstantValue = constantValue;
                 Method = method;
                 ConstrainedToType = constrainedToType;
                 OriginalUserDefinedOperatorsOpt = originalUserDefinedOperatorsOpt;
+                IsUnconvertedInterpolatedStringAddition = isUnconvertedInterpolatedStringAddition;
+                InterpolatedStringHandlerData = interpolatedStringHandlerData;
```

Comment:
Is it the case that we'll have `InterpolatedStringHandlerData` only after the conversion occurs (ie. when `IsUnconvertedInterpolatedStringAddition` is false)?
Consider asserting any relationship between `IsUnconvertedInterpolatedStringAddition` and `InterpolatedStringHandlerData`.